src/jsonpath/parser.c

```diff
@@ -147,14 +154,13 @@ static bool parse_filter_list(PARSER_PARAMS, struct ast_node* tok) {
         zend_throw_exception(spl_ce_RuntimeException, "Array slice indexes must be integers", 0);
         return false;
```

Comment:
I get a memory leak in tests/015.phpt (Test array slice with strings as start, end, and step), which I managed to fix in the node pool PR with:

```
if (sep_found == AST_INDEX_SLICE) {
  zend_throw_exception(spl_ce_RuntimeException, "Array slice indexes must be integers", 0);

  /* If we end up here, we may already have parsed one or more string literals into the */
  /* hash table under the assumption that we were dealing with an index filter. Free those. */
  if (tok->data.d_nodes.ht != NULL) {
    zend_array_destroy(tok->data.d_nodes.ht);
    tok->data.d_nodes.count = 0;
  }

  return false;
}
```

Something like that perhaps?

Reply:
Thanks for flagging! I made a go at unifying the data structure for slices, index lists, and node selector lists using a HashTable, which incidentally solve this leak. Check out fd756e7560d4815ed78fcc03ad2a382c03751dd0.